22-discovery.ks

```diff
@@ -48,7 +48,10 @@ systemctl disable ipmi.service
 echo " * open foreman-proxy port via firewalld"
 firewall-offline-cmd --zone=public --add-port=8443/tcp --add-port=8448/tcp
 
-echo " * setting up foreman proxy"
+echo " * setting up foreman proxy service"
+sed -i 's/After=.*/After=basic.target network-online.target nm-prepare.service/' /usr/lib/systemd/system/foreman-proxy.service
+sed -i 's/Wants=.*/Wants=basic.target network-online.target nm-prepare.service/' /usr/lib/systemd/system/foreman-proxy.service
+sed -i '/\[Unit\]/a ConditionPathExists=/etc/NetworkManager/system-connections/primary' /usr/lib/systemd/system/foreman-proxy.service
```

Comment:
+1 - I just tested this part of code in our environment and it works (Dell PowerEdge R730, 9 NICs in each) - discovery image registers servers via foreman-proxy and foreman reboots and deploys servers automatically.